libbeat/cmd/instance/beat.go

```diff
@@ -984,6 +985,64 @@ func (b *Beat) createOutput(stats outputs.Observer, cfg config.Namespace) (outpu
 		return outputs.Group{}, nil
 	}
 
+	// TODO (Tiago): find a proper name for this logger
+	logger := logp.L().Named("ssl.cert.reloader")
+	// Here the output is created and we have acces to the Beat struct (with the manager)
+	// as a workaround we can unpack the new settings and trigger the reload-watcher from here
+
+	// We get an output config, so we extract the 'SSL' bit from it
+	rawTLSCfg, err := cfg.Config().Child("ssl", -1)
+	if err != nil {
+		return outputs.Group{}, fmt.Errorf("could not extract the 'ssl' section of the output config: %w", err)
```

Comment:
do we get an error if the config doesn't have an ssl section?, just want to make sure we can still get an output even if it doesn't have ssl.

Reply:
That's one of the many parts missing, aside the happy path, I haven't really handled anything else.